include/view/opengl_interfacing/camera.hpp

```diff
@@ -6,26 +6,38 @@
 
 class Camera
 {
-private:
+protected:
 	glm::vec3 position;
 	glm::vec3 direction;
+	glm::vec3 forward;
 	glm::vec3 up;
 
 	glm::mat4 viewMatrix;
 	glm::mat4 projectionMatrix;
-protected:
 public:
 	//all angles in radians
 	Camera(float fov, float aspectRatio, float nearDraw, float farDraw);
 
 	const glm::mat4 &getViewMatrix() const;
 	const glm::mat4 &getProjectionMatrix() const;
 
-	//this only sets the position vector! this does not change direction where camera looks at, meaning that
-	//using only this will result in camera moving, but looking at the point set in direction vector
+	const glm::vec3 &getPosition() const;
+	const glm::vec3 &getDirection() const;
+	const glm::vec3 &getForward() const;
+	const glm::vec3 &getUp() const;
+
 	void setPosition(const glm::vec3 &position);
+	void setDirection(const glm::vec3 &direction, const glm::vec3 &up);
+
+	virtual void moveForward() = 0;
+	virtual void moveBackward() = 0;
+	virtual void moveLeft() = 0;
+	virtual void moveRight() = 0;
+	virtual void moveUp() = 0;
+	virtual void moveDown() = 0;
+	//heads up: here will almost certainly be rotation methods like the ones above
 
-	void setDirection(const glm::vec3 &direction);
+	virtual void move() = 0;
```

Comment:
I don't think that this function is necessary, I added all the cardinal directions above, and unfortunately we cannot easily contain each direction in a single function (e.g. horizontally, and giving a bool to determine left or right). The reason is simple, up and down or forward and backward can mean different things for different types of cameras. Please use the methods I have added above and remove this one.